diffus/src/lcs.rs

```diff
@@ -28,12 +28,32 @@ where
     I: DoubleEndedIterator<Item = T>,
     J: DoubleEndedIterator<Item = T>,
 {
-    let prefix_eq = x().zip(y()).take_while(|(x, y)| x.same(y)).count();
-    let suffix_eq = x()
-        .rev()
-        .zip(y().rev())
-        .take_while(|(x, y)| x.same(y))
-        .count();
+    let mut prefix_eq = 0;
```

Comment:
Something like
```rust
let mut x_iter = x();
let mut y_iter = y();
let prefix_eq = x_iter
    .by_ref()
    .zip(y_iter.by_ref())
    .take_while(|(x, y)| x.same(y))
    .count();
let check_suffix = x_len.min(y_len) != prefix_eq;
let suffix_eq = if check_suffix {
    x_iter
        .rev()
        .zip(y_iter.rev())
        .take_while(|(x, y)| x.same(y))
        .count()
} else {
    0
};
```
to avoid the amount of mutating fields

Reply:
Fixed.